test/k8sT/istio.go

```diff
@@ -0,0 +1,276 @@
+// limitations under the License.
+
+package k8sTest
+
+import (
+	"fmt"
+
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/gomega"
+)
+
+// This tests the Istio 1.0.0 integration, following the configuration
+// instructions specified in the Istio Getting Started Guide in
+// Documentation/gettingstarted/istio.rst.
+// Changes to the Getting Started Guide may require re-generating or copying
+// the following manifests:
+// - istio-cilium.yaml
+// - bookinfo-v1-istio.yaml
+// - bookinfo-v2-istio.yaml
+// - istio-sidecar-init-policy.yaml
+// Cf. the comments below for each manifest.
+var _ = Describe("K8sIstioTest", func() {
+
+	var (
+		// istioSystemNamespace is the default namespace into which Istio is
+		// installed.
+		istioSystemNamespace = "istio-system"
+
+		// istioYAMLPath is the istio-cilium.yaml file generated following the
+		// instructions in Documentation/gettingstarted/istio.rst to setup
+		// Istio 1.0.0. mTLS is enabled.
+		istioYAMLPath = helpers.ManifestGet("istio-cilium.yaml")
+
+		// wgetCommand is the command used in this test because the Istio apps
+		// do not provide curl.
+		wgetCommand = fmt.Sprintf("wget --tries=2 --connect-timeout %d", helpers.CurlConnectTimeout)
+
+		kubectl       *helpers.Kubectl
+		ciliumPodK8s1 string
+	)
+
+	BeforeAll(func() {
+		k8sVersion := helpers.GetCurrentK8SEnv()
+		if k8sVersion == "1.7" {
+			Skip(fmt.Sprintf("Istio doesn't support K8S %s", k8sVersion))
+		}
+
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+
+		err := kubectl.CiliumInstall(helpers.CiliumDSPath)
+		Expect(err).To(BeNil(), "Cilium cannot be installed")
+
+		ExpectCiliumReady(kubectl)
+		ExpectKubeDNSReady(kubectl)
+
+		ciliumPodK8s1, err = kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+		Expect(err).Should(BeNil(), "Cannot get cilium pod on k8s1")
+
+		By("Creating the istio-system namespace")
+		res := kubectl.NamespaceCreate(istioSystemNamespace)
+		res.ExpectSuccess("unable to create namespace %q", istioSystemNamespace)
+
+		By("Creating the Istio resources")
+		res = kubectl.Apply(istioYAMLPath)
+		res.ExpectSuccess("unable to create Istio resources")
+
+		// Ignore one-time jobs and Prometheus. All other pods in the
+		// namespaces have an "istio" label.
+		err = kubectl.WaitforPods(istioSystemNamespace, "-l istio", 300)
+		Expect(err).To(BeNil(),
+			"Istio pods are not ready after timeout in namespace %q", istioSystemNamespace)
+	})
+
+	AfterAll(func() {
+		By("Deleting the Istio resources")
+		_ = kubectl.Delete(istioYAMLPath)
+
+		By("Deleting the istio-system namespace")
+		_ = kubectl.NamespaceDelete(istioSystemNamespace)
+	})
+
+	JustAfterEach(func() {
+		kubectl.ValidateNoErrorsOnLogs(CurrentGinkgoTestDescription().Duration)
+	})
+
+	AfterFailed(func() {
+		kubectl.CiliumReport(helpers.KubeSystemNamespace,
+			"cilium endpoint list",
+			"cilium bpf proxy list")
```

Comment:
Is the `bpf proxy list` command good in this case?

Reply:
The output is interesting, but only while a connection is established. It would show whether any connection is redirected to the node Envoy proxy. But I agree that after the test finishes, that table is always in any case, so I've removed it.
I will find another way to check whether the node Envoy proxy is started.